app/routers/schema.py

```diff
@@ -0,0 +1,163 @@
+        await aries_agent_controller.terminate()
+        return created_schemas
+
+    except Exception as e:
+        await aries_agent_controller.terminate()
+        raise HTTPException(
+            status_code=418,
+            detail=f"Something went wrong.\n Could not get schema from ledger.\n{e}.",
+        )
+
+
+@router.get("/schema/schema_definition", tags=["schema", "credential"])
+async def schema_define():
+    """
+    Define Schema
+    """
+    return {"msg": "from schema define"}
+
+
+@router.get("/schema/schema_define_getter", tags=["schema", "credential"])
+async def schema_define_getter():
+    """
+    Define Schema
+    """
+    final = await schema_define()
+    return final
+
+
+@router.post(
+    "/schema/write-schema-and-credential-definition", tags=["schema", "credential"]
+)
+async def write_credential_schema(
+    schema_name: str, schema_version: str, schema_attrs: List[str] = Query(None)
+):
+    """
+    Create schema and credential definition and
+    write it to the ledger.
+
+    Parameters:
+    ----------
+    * schema_name: str
+        The name of the schema to be defined
+    * schema_version: str
+        The version of the schema to be written\n
+        Should be of the form x.x.x where x is an integer
+    * schema_attributes: list, optional
+        A list of attributes for the schema (default is None)
+
+    Returns:
+    --------
+    as json:
+    * schema
+    * schema_id
+    * credential_definition
+    * credential_id
+    """
+    try:
+        aries_agent_controller = aries_cloudcontroller.AriesAgentController(
+            admin_url=f"{admin_url}:{admin_port}",
+            api_key=f"{admin_api_key}",
+            is_multitenant=is_multitenant,
+        )
+
+        # Defining schema and writing it to the ledger
+
+        schema_definition_request = SchemaDefinitionRequest(
+            schema_name=schema_name,
+            schema_version=schema_version,
+            schema_attributes=schema_attrs,
+        ).dict()
+
+        write_schema_resp = await aries_agent_controller.schema.write_schema(
+            schema_definition_request.schema_name,
+            schema_definition_request.schema_attributes,
+            schema_definition_request.schema_version,
+        )
+
+        if not write_schema_resp or write_schema_resp == {}:
+            await aries_agent_controller.terminate()
+            raise HTTPException(
+                status_code=418,
+                detail=f"Something went wrong.\n Could not write schema to ledger.\n{schema}",
+            )
+        schema_id = write_schema_resp["schema_id"]
+
+        # Writing credential definition
+        credential_definition = await aries_agent_controller.definitions.write_cred_def(
+            schema_id
+        )
+        if not credential_definition:
+            await aries_agent_controller.terminate()
+            raise HTTPException(
+                status_code=418,
```

Comment:
Same here 404 or 500 ( I reckon 404). 

Can you please go over all of them? I don't think 418 should be raised really unless for the dummy response in admin because that's simply not developed yet